addons/web/static/tests/views/list/list_view.test.js

```diff
@@ -4933,6 +4933,65 @@ test(`date field aggregates in grouped lists`, async () => {
     ]);
 });
 
+test(`when a single non empty group is emptied, dont display its group header`, async () => {
```

Comment:
I don't really understand the test. I think it could be simpler and easier to read/understand by replicating a similar scenario as the Post case of accounting: have a group header button that calls a method that deletes all records of the group and click on it. You don't need to override web_read_group I think, nor to add a new selection field. You can set configure the view with a groupBy and a domain s.t. there's a single group.

I'm may be missing something, in which case don't hesitate to reach out to me.